derivation.nix

```diff
@@ -70,15 +107,35 @@ in
           buildPhase = ''
             runHook preBuild
 
-            pnpm run ${script}
+            ${if isWorkspace
+              then ''
+                ${forEachComponent (component: ''
+                  (cd "${component}" && pnpm run ${script})
```

Comment:
Looking at this loop, and the loop above, it may be better to define a computed components that is `["."]` for the non-workspace case. This would remove some of the `if isWorkspace` tests.

Reply:
I decided against this. Shorter code is not worth conflating components and non-components builds. I did however pre-compute all of the things that need to be different for a workspace build, many of them into lists, and then the derivation just loops over those lists.